src/components/DescriptionView.tsx

```diff
@@ -0,0 +1,237 @@
+    data: {
+      message: '',
+      errorDetail: `Exceeds ${MAX_LENGTH} characters.  Please revise and save again.`,
+    },
+  },
+};
+
+type DescriptionViewProps = {
+  classes: ClassNameMap;
+  description: string;
+  updateDescriptionFn: any;
+  userRoles: Array<string>;
+};
+
+type DescriptionViewState = {
+  hasDescriptionChanged: boolean;
+  descriptionValue: string | null;
+  isEditing: boolean;
+  isPendingSave: boolean;
+};
+
+const initialState: DescriptionViewState = {
+  hasDescriptionChanged: false,
+  descriptionValue: null,
+  isEditing: false,
+  isPendingSave: false,
+};
+
+class DescriptionView extends React.PureComponent<DescriptionViewProps, DescriptionViewState> {
+  constructor(props: DescriptionViewProps) {
+    super(props);
+    initialState.descriptionValue = props.description;
+    this.state = initialState;
+    this.textFieldRef = React.createRef();
+  }
+
+  componentDidUpdate() {
+    const { descriptionValue, isPendingSave } = this.state;
+    const { description } = this.props;
+    if (isPendingSave && description === descriptionValue) {
+      this.setState({ hasDescriptionChanged: false });
+      this.setState({ isPendingSave: false });
+      this.onExitEdit();
+    }
+  }
+
+  textFieldRef: React.RefObject<any>;
+
+  descriptionChanged(newDescription: string | null, originalDescription: string | null) {
+    if (newDescription && newDescription.length > MAX_LENGTH) {
+      showNotification(descriptionTooLongError);
+      this.setState({ descriptionValue: newDescription.substring(0, 2046) });
+    } else {
+      this.setState({ descriptionValue: newDescription });
+    }
+    if (newDescription !== originalDescription) {
+      this.setState({ hasDescriptionChanged: true });
+    } else {
+      this.setState({ hasDescriptionChanged: false });
+    }
+  }
+
+  onDescriptionTextBlur(newDescription: string | null, originalDescription: string | null) {
+    this.descriptionChanged(newDescription, originalDescription);
+    this.onExitEdit();
+  }
+
+  onDescriptionTextClick() {
+    const { isEditing } = this.state;
+    if (!isEditing) {
+      this.setState({ isEditing: true });
+    }
+  }
+
+  onEditClick() {
+    const { isEditing } = this.state;
+    if (!isEditing) {
+      this.textFieldRef.current.focus();
+      this.setState({ isEditing: true });
+    }
+  }
+
+  onExitEdit() {
+    const { description } = this.props;
+    const { descriptionValue } = this.state;
+    if (description === descriptionValue) {
+      this.setState({ isEditing: false });
+    }
+  }
+
+  onSaveClick(descriptionText: string | null) {
+    const { updateDescriptionFn } = this.props;
+    this.setState({ isPendingSave: true });
+    updateDescriptionFn(descriptionText);
+  }
+
+  render() {
+    const { classes, description, userRoles } = this.props;
+    const { hasDescriptionChanged, descriptionValue, isEditing } = this.state;
+    const canEdit = userRoles.includes(DatasetRoles.STEWARD);
+
+    return (
+      <>
```

Comment:
oh interesting: I've never seen this done with the empty bracket rather than arrays.  I like this